nbdime/merging/notebooks.py

```diff
@@ -71,13 +72,31 @@ def decide_notebook_merge(base, local, remote, args=None):
     local_diffs = diff_notebooks(base, local)
     remote_diffs = diff_notebooks(base, remote)
 
+    if args and args.log_level == "DEBUG":
+        print("="*70)
+        print("In merge, base-local diff:")
+        pretty_print_notebook_diff("<base>", "<local>", base, local_diffs)
+        print("="*70)
+        print("In merge, base-remote diff:")
+        pretty_print_notebook_diff("<base>", "<remote>", base, remote_diffs)
+
```

Comment:
Why not create a module level logger (`_logger = logging.getLogger(__name__)`) and use `_logger.debug(things_to_print)` ? You could pass a stringio as `out` argument, and then `_logger.debug(buf.getvalue())`. This will ensure it goes the right place, even if using a log file.